library/std/src/sys/sync/condvar/no_threads.rs

```diff
@@ -1,25 +1,50 @@
+use crate::cell::RefCell;
 use crate::sys::sync::Mutex;
+use crate::thread::sleep;
 use crate::time::Duration;
 
-pub struct Condvar {}
+pub struct Condvar {
+    value: RefCell<u32>,
+}
+
+unsafe impl Send for Condvar {}
+unsafe impl Sync for Condvar {} // no threads on this platform
 
 impl Condvar {
     #[inline]
     pub const fn new() -> Condvar {
-        Condvar {}
+        Condvar { value: RefCell::new(0) }
     }
 
     #[inline]
-    pub fn notify_one(&self) {}
+    pub fn notify_one(&self) {
+        *self.value.borrow_mut() += 1;
+    }
 
     #[inline]
-    pub fn notify_all(&self) {}
+    pub fn notify_all(&self) {
+        *self.value.borrow_mut() += 1;
+    }
 
     pub unsafe fn wait(&self, _mutex: &Mutex) {
-        panic!("condvar wait not supported")
+        let mut value = self.value.borrow_mut();
+        if *value == 0 {
+            // Since the target does not have threads,
+            // no notification can ever arrive.
+            loop {}
```

Comment:
Most wasm primitives panic when they hit a situation that would otherwise sleep forever.

Reply:
I changed it to a panic.